source/docs/zero-to-robot/step-4/creating-benchtop-test-program-cpp-java.rst

```diff
@@ -84,27 +115,73 @@ Defining the variables for our sample robot
+               :language: java
+               :lines: 19-36
+               :linenos:
+               :lineno-start: 19
 
-      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/v2022.4.1/wpilibcExamples/src/main/cpp/examples/GettingStarted/cpp/Robot.cpp
-         :language: c++
-         :lines: 49-57
-         :linenos:
-         :lineno-start: 49
+         .. group-tab:: C++
+
+            .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/v2022.4.1/wpilibcExamples/src/main/cpp/examples/GettingStarted/cpp/Robot.cpp
+               :language: c++
+               :lines: 12-20
+               :linenos:
+               :lineno-start: 12
+
+            .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/v2022.4.1/wpilibcExamples/src/main/cpp/examples/GettingStarted/cpp/Robot.cpp
+               :language: c++
+               :lines: 49-57
+               :linenos:
+               :lineno-start: 49
+
+   .. group-tab:: CTRE
+
+      .. tabs::
+
+         .. group-tab:: Java
+
+            .. code-block:: java
+
+               public class Robot extends TimedRobot {
+                  private final WPI_TalonFX m_leftDrive = new WPI_TalonFX(0);
+                  private final WPI_TalonFX m_rightDrive = new WPI_TalonFX(1);
+                  private final DifferentialDrive m_robotDrive = new DifferentialDrive(m_leftDrive, m_rightDrive);
+                  private final Joystick m_stick = new Joystick(0);
+                  private final Timer m_timer = new Timer();
+
+         .. group-tab:: C++
+
+            Inside of ``Robot.h``
+
+            .. code-block:: cpp
+
+               public:
+                Robot() {
+                   m_right.SetInverted(true);
+                   m_robotDrive.SetExpiration(100_ms);
+                   // We need to invert one side of the drivetrain so that positive voltages
+                   // result in both sides moving forward. Depending on how your robot's
+                   // gearbox is constructed, you might have to invert the left side instead.
+                   m_timer.Start();
+                }
+
+            .. code-block:: cpp
+
+               private:
+                // Robot drive system
+                ctre::phoenix::motorcontrol::can::WPI_TalonFX m_left{0};
+                ctre::phoenix::motorcontrol::can::WPI_TalonFX m_right{1};
+                frc::DifferentialDrive m_robotDrive{m_left, m_right};
+
+                frc::Joystick m_stick{0};
+                frc::Timer m_timer;
 
 The sample robot in our examples will have a joystick on USB port 0 for arcade drive and two motors on PWM ports 0 and 1. Here we create objects of type DifferentialDrive (m_robotDrive), Joystick (m_stick) and Timer (m_timer). This section of the code does three things:
```

Comment:
This needs update for different speed controller options

Reply:
I do not understand. I do not plan to talk about the individual characteristics (APIs) of each speed controller.

Reply:
Is specified PWM, which is no longer true in all cases